config/initializers/session_store.rb

```diff
@@ -1,3 +1,11 @@
 # Be sure to restart your server when you modify this file.
 
-Rails.application.config.session_store :cookie_store, key: "_PracticalDeveloper_session"
+# we want a default in case the expiration is not set or set to 0
+# because 0 is an invalid value
+app_config_expires_in = ApplicationConfig["SESSION_EXPIRY_SECONDS"].to_i
+expires_in = app_config_expires_in.positive? ? app_config_expires_in : 2.weeks.to_i
+
+Rails.application.config.session_store :redis_store,
+                                       key: ApplicationConfig["SESSION_KEY"],
+                                       servers: ApplicationConfig["REDIS_URL"],
+                                       expires_in: expires_in
```

Comment:
We probably want to add a couple more [session_store options](https://github.com/redis-store/redis-rails#session-storage) here such as `signed: true`(I have never used this option before but seems like a good idea) and `secure: Rails.env.production?`. I am seeing `httponly: true` in some of my past config files but it is not mentioned in the docs, however it [indeed is an option](https://github.com/redis-store/redis-actionpack/blob/6c3af02a0b184651316cb89c3e563ee18d0f0389/lib/action_dispatch/middleware/session/redis_store.rb#L38) and this is the [PR that added it](https://github.com/redis-store/redis-actionpack/pull/17) so that might be worth it to add as well for security.

Reply:
You want http only set to true so the cookie can’t be read via JavaScript. For more info, see https://www.owasp.org/index.php/HttpOnly

Reply:
Done :)

![Screenshot 2019-10-23 at 6 06 42 PM](https://user-images.githubusercontent.com/146201/67412830-4fa12e80-f5c0-11e9-94e6-88e30bddc19e.png)

I also changed `expires_in` which looks like it's been deprecated in favor of `expires_after`